src/apps/search-header/helpers.ts

```diff
@@ -74,4 +81,84 @@ export function isDisplayedAsWorkSuggestion(
   return Boolean(dataWithWorkId.length);
 }
 
+enum SearchType {
+  Creator = "creator",
+  Subject = "subject",
+  Dk5 = "dk5",
+  General = "q"
+}
+
+interface SearchContext {
+  searchType: SearchType;
+  initialQuery: string;
+  creatorFilter: string | null;
+  subjectFilter: string | null;
+  dk5Filter: string | null;
+}
+
+export function getSearchContextFromUrl(): SearchContext {
+  const qParam = getUrlQueryParam("q");
+  const creatorFilterParam = getUrlQueryParam("creators");
+  const subjectFilterParam = getUrlQueryParam("subjects");
+  const dk5FilterParam = getUrlQueryParam("dk5");
```

Comment:
You could also create the object immediately and vary it for the different cases:
```typescript
const searchContext = {
  searchType: SearchType.General,
  initialQuery: getUrlQueryParam("q") || "",
  creatorFilter: getUrlQueryParam("creators"),
  subjectFilter: getUrlQueryParam("subjects"),
  dk5Filter: getUrlQueryParam("dk5"),
};

if (creatorFilterParam) {
  return {
    ...searchContext,
    searchType: SearchType.Creator,
    initialQuery: searchContext.creatorFilter,
  };
}
...